daml/Marketplace/Trading/TradeInstruction.daml

```diff
@@ -0,0 +1,137 @@
+module Marketplace.Trading.TradeInstruction where
+
+import DA.Assert ((===))
+import DA.Finance.Asset (AssetDeposit, AssetDeposit_Split(..))
+import DA.Finance.Trade.SettlementInstruction (SettlementInstruction(..), SettlementInstruction_Process(..), SettlementDetails(..))
+import DA.Finance.Trade.Dvp (Dvp(..), SettlementStatus(..))
+import DA.Finance.Types (Id(..), Asset(..), Account(..), MasterAgreement(..))
+import DA.Finance.Utils (assertOnOrAfterDateMsg, zipChecked, fetchAndArchive)
+import DA.List (head, last)
+import DA.Next.Set (Set, empty, insert, fromList)
+import DA.Optional (whenSome)
+import Marketplace.Listing (Listing)
+import Marketplace.Trading.Order qualified as Order
+
+data CustomerAccounts = CustomerAccounts with
+    customer : Party
+    allocationAccount : Account
+    tradingAccount : Account
+  deriving (Eq, Show)
+
+type T = TradeInstruction
+
+template TradeInstruction
```

Comment:
So what's the difference between a `TradeInstruction` and a `Trade`?

Reply:
We should rename this - TradeInstruction generates the DvP, Settlement instructions, updates the order, etc. but Trade has the Settle choice to execute the DvP. I'll speak with Georg to rename this.